diff -urN libowfat-0.29/GNUmakefile libowfat-0.29-patched/GNUmakefile
--- libowfat-0.29/GNUmakefile	2012-04-18 08:26:31.000000000 +1100
+++ libowfat-0.29-patched/GNUmakefile	2014-03-12 21:37:23.709442932 +1100
@@ -13,12 +13,12 @@
 buffer.a mmap.a taia.a tai.a dns.a case.a mult.a array.a io.a \
 textcode.a cdb.a
 
-all: $(LIBS) libowfat.a libsocket t
+all: $(LIBS) libowfat.so libsocket t
 
 CROSS=
 #CROSS=i686-mingw-
 CC=$(CROSS)gcc
-CFLAGS=-pipe -W -Wall -O2 -fomit-frame-pointer
+CFLAGS=-pipe -W -Wall -O2 -fomit-frame-pointer -fPIC
 #CFLAGS=-pipe -Os -march=pentiumpro -mcpu=pentiumpro -fomit-frame-pointer -fschedule-insns2 -Wall
 
 # CFLAGS += -fstrict-aliasing -Wstrict-aliasing=2
@@ -134,9 +134,8 @@
 $(TAIA_OBJS) $(TAI_OBJS) $(CASE_OBJS) $(ARRAY_OBJS) $(MULT_OBJS) \
 $(IO_OBJS) $(CDB_OBJS)
 
-libowfat.a: $(ALL_OBJS)
-	$(CROSS)ar cr $@ $(ALL_OBJS)
-	-$(CROSS)ranlib $@
+libowfat.so: $(ALL_OBJS)
+	$(CROSS)gcc -shared -Wl,-soname,libowfat.so.$(MAJOR) -o $@.$(VERSION) $(ALL_OBJS)
 
 CFLAGS+=-I.
 
@@ -153,8 +152,8 @@
   iarray.h io_internal.h haveepoll.h havekqueue.h havedevpoll.h \
   havesigio.h CAS.h
 
-t: t.o libowfat.a libsocket
-	$(DIET) $(CC) -g -o $@ t.o libowfat.a `cat libsocket` -lpthread
+t: t.o libowfat.so libsocket
+	$(DIET) $(CC) -g -o $@ t.o libowfat.so.$(VERSION) `cat libsocket` -lpthread
 
 .PHONY: all clean tar install rename
 clean:
@@ -168,13 +167,17 @@
 openreadclose.h readclose.h ndelay.h array.h io.h safemult.h iob.h havealloca.h \
 errmsg.h cdb.h cdb_make.h rangecheck.h iarray.h va_narg.h isset.h
 
+VERSION=$(shell head -n 1 CHANGES|sed 's/://')
+MAJOR=$(shell echo $(VERSION)|sed 's/\..*//')
+CURNAME=$(notdir $(shell pwd))
+
 install-inc:
 	install -d $(INCLUDEDIR)
 	install -m 644 $(INCLUDES) $(INCLUDEDIR)
 
-install-lib: libowfat.a
+install-lib: libowfat.so
 	install -d $(LIBDIR)
-	install -m 644 libowfat.a $(LIBDIR)
+	install -m 755 libowfat.so.$(VERSION) $(LIBDIR)
 
 install-man:
 	install -d $(MAN3DIR)
@@ -185,10 +188,7 @@
 uninstall:
 	rm -f $(patsubst %.h,$(INCLUDEDIR)/%.h,$(INCLUDES))
 	rm -f $(patsubst %.3,$(MAN3DIR)/%.3,$(notdir $(wildcard */*.3)))
-	rm -f $(LIBDIR)/libowfat.a
-
-VERSION=libowfat-$(shell head -n 1 CHANGES|sed 's/://')
-CURNAME=$(notdir $(shell pwd))
+	rm -f $(LIBDIR)/libowfat.so.$(VERSION)
 
 tar: Makefile clean rename
 	rm -f dep libdep
